### PR TITLE
[Rasa] Upgrade Rasa Fork Dependencies

### PR DESCRIPTION
**Pull request by giancaire**
**Description**:

This PR upgrades several dependencies to address various vulnerability alerts:

`cryptography`
`aiohttp`
`sentry-sdk`
`pydantic`

Please note that upgrading `keras` and `tensorflow` to their next versions may introduce breaking changes. Further assessment and adjustments will be necessary for these dependencies.

**Ticket**
[[Rasa] Upgrade Rasa Fork Dependencies](https://cairehealth.atlassian.net/jira/software/projects/CRE/boards/1?selectedIssue=CRE-2714)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,345 +1,183 @@
...
-absl-py = ">=0.9,<1.5"
-apscheduler = ">=3.6,<3.11"
-tqdm = "^4.31"
-networkx = ">=2.4,<3.2"
-fbmessenger = "~6.0.0"
-pykwalify = ">=1.7,<1.9"
-coloredlogs = ">=10,<16"
-"ruamel.yaml" = ">=0.16.5,<0.17.22"
+python = ">=3.10, <3.11"
+rasa-sdk = ">=3.6.2"
```
Comment (werkshy) [inline pyproject.toml:40]: Should we be using `"^3.6.2"` for the deps? Otherwise we're allowing major version updates which are quite likely to cause issues. See https://python-poetry.org/docs/dependency-specification/

Comment (giancaire) [reply pyproject.toml:40]: Yes, that's great feedback! I'll go back through and add the caret (^) version specifiers before merging to avoid unintentional major version updates. Thanks for pointing that out!

File: poetry.lock
Comment (werkshy) [inline poetry.lock:1]: Great, only 4k diff! (was 8k!) 😅

**Review by werkshy — APPROVED** [review]
The lockfile diff is just too big to review - happy to rely on "does it work" and "does it make dependabot shut up" as our main metric.

pyproject.toml looks good 👍
